dotCMS/src/main/java/com/dotmarketing/portlets/templates/design/bean/TemplateLayoutColumn.java

```diff
@@ -1,50 +1,54 @@
 package com.dotmarketing.portlets.templates.design.bean;
 
 import com.dotmarketing.portlets.templates.design.util.PreviewTemplateUtil;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
 
 import java.util.List;
 
 /**
  * Created by Jonathan Gamba
  * Date: 10/25/12
  */
-public class TemplateLayoutColumn {
+public class TemplateLayoutColumn extends ContainerHolder {
 
-    public static String TYPE_BODY = "body";//Main body column
-    public static String TYPE_COLUMN = "column";//Normal column inside a row
-    public static String TYPE_SIDEBAR = "sidebar";//Sidebar column
+    private Integer widthPercent;
+    private Integer width;
+    private int leftIndex;
 
-    //ONLY FOR SIDEBARS!!!
-    public static String LOCATION_RIGHT = "right";
-    public static String LOCATION_LEFT = "left";
-    //ONLY FOR SIDEBARS!!!
+    @JsonCreator
+    public TemplateLayoutColumn(@JsonProperty("containers") List<String> containers,
+                                @JsonProperty("widthPercent") final int widthPercent,
+                                @JsonProperty("leftIndex") final int leftIndex) {
+        super(containers);
 
-    public boolean preview;
-    public String type;
-    public String location;//ONLY FOR SIDEBARS!!!
-    public List<String> containers;
-    public Integer widthPercent;
-    public Integer width;
-
-    public List<TemplateLayoutRow> rows;
-
-    public List<String> getContainers () {
-        return containers;
-    }
-
-    public void setContainers ( List<String> containers ) {
-        this.containers = containers;
-    }
-
-    public boolean isPreview () {
-        return preview;
-    }
-
-    public void setPreview ( boolean preview ) {
-        this.preview = preview;
+        this.widthPercent = widthPercent;
+        this.leftIndex = leftIndex;
     }
 
     public Integer getWidthPercent () {
+        if (widthPercent == null || widthPercent == 0){
+            switch (this.width) {
+                case 12:
+                    this.widthPercent = 100;
+                    break;
+                case 9:
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 9 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS109)